pkg/cloudprovider/servicecontroller/servicecontroller.go

```diff
@@ -465,3 +510,72 @@ func hostsFromNodeList(list *api.NodeList) []string {
 	}
 	return result
 }
+
+// watchNodes handles updating the hosts pointed to by all external load
+// balancers whenever the set of nodes in the cluster changes.
+func (s *ServiceController) watchNodes(nodeLister *cache.StoreToNodeLister, period time.Duration) {
```

Comment:
Is there not an actual "Watch" we can perform on all of the nodes instead of polling like this? Maybe something like we currently do in the kubelet using the a cache.ListWatch (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubelet/kubelet.go#L235-L244).